### PR TITLE
Added missing interface for CL_SEMAPHORE_DEVICE_HANDLE_LIST_KHR property

### PR DESCRIPTION
**Pull request by shajder**
Completed during work on [CTS PR](https://github.com/KhronosGroup/OpenCL-CTS/pull/1942)

### CODE REVIEW

**Comment by shajder** [comment]
I see the error from windows build:

```
       (Link target) -> 
         emulate.obj : error LNK2019: unresolved external symbol _clGetDeviceInfo@20 referenced in function "public: static struct _cl_semaphore_khr * __cdecl _cl_semaphore_khr::create(struct _cl_context *,unsigned __int64 const *,int *)" (?create@_cl_semaphore_khr@@SAPAU1@PAU_cl_context@@PB_KPAH@Z) [C:\projects\simpleopenclsamples\build\layers\11_semaemu\SemaEmu.vcxproj]
         C:\projects\simpleopenclsamples\build\layers\11_semaemu\Release\SemaEmu.dll : fatal error LNK1120: 1 
```

I am calling plain `clGetDeviceInfo` because `g_pNextDispatch->clGetDeviceInfo` for some reason does not use `clGetDeviceInfo_override`. Any clue on your mind how to call `clGetDeviceInfo` at `Semaphore::create` level ? Thanks

File: layers/11_semaemu/emulate.cpp [hunk 1]
```diff
@@ -91,6 +94,38 @@ typedef struct _cl_semaphore_khr
...
+                for (auto sema_type : types) {
+                  if (type == sema_type) {
+                    capable = true;
+                    break;
+                  }
+                }
+
+                if (!capable) {
+                  errorCode = CL_INVALID_DEVICE;
+                  break;
```
[the remark this answers is not in the document]
Comment (shajder) [reply layers/11_semaemu/emulate.cpp]: Done, I put `isDeviceWithinContext` procedure in `emulate.cpp` but if you think there is better location just let me know.

**Review by bashbaug — APPROVED** [review]
Thanks!  The latest version LGTM.

**Comment by bashbaug** [comment]
I'm going to go ahead and merge this even though there are CI failures because I think the failures are unrelated and already fixed in the main branch.

**Pull request merged**